staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go

```diff
@@ -230,8 +230,7 @@ func (s preparedGenericAPIServer) NonBlockingRun(stopCh <-chan struct{}) error {
 
 	s.RunPostStartHooks()
 
-	// err == systemd.SdNotifyNoSocket when not running on a systemd system
-	if err := systemd.SdNotify("READY=1\n"); err != nil && err != systemd.SdNotifyNoSocket {
+	if _, err := systemd.SdNotify(true, "READY=1\n"); err != nil {
```

Comment:
I assume we don't call NonBlockingRun multiple times. Not sure what would happen if you did. @deads2k ?